### PR TITLE
Introduce Deltalake

### PR DESCRIPTION
**Pull request by mrocklin**
This doesn't work yet.  I'm getting odd column errors in the tpch query.  Sharing early in case people have thoughts.

### CODE REVIEW

File: generate_data.py [hunk 1]
```diff
@@ -34,6 +35,9 @@ def generate(scale: float, path: os.PathLike) -> None:
                 SET s3_session_token='{creds.token}';
                 """
             )
+        else:
+            path = pathlib.Path(path)
+            path.mkdir(parents=True, exist_ok=True)
```
Comment (jrbourbeau) [inline generate_data.py:40]: Hmm I'm curious why this is needed given the `mkdir` calls later in this function, but I'll probably find out while running this code locally. Feel free to fill me in if you recall why this was added

**Review by jrbourbeau — COMMENTED** [review]
Thanks @mrocklin I'll try this out and try to get it running on the cloud as well

File: generate_data.py (hunk 1 above)
Comment (mrocklin) [reply generate_data.py:40]: This might just be a merge issue.

**Comment by jrbourbeau** [comment]
@mrocklin I'm stepping away for a bit but will circle back here. Just pushed commits that resolve merge conflicts and make this run on S3 (and local still)

**Review by jrbourbeau — APPROVED** [review]
Thanks @mrocklin

**Pull request merged**